sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/AcrClientCredentials.cs

```diff
@@ -53,73 +56,63 @@ public enum LoginMode
         private AuthToken _aadAccess;
 
         #endregion
+        
+        // initialization logic helper.
+        private void Initialize()
+        {
+            if (_mode == LoginMode.Basic) // Basic Authentication
+            {
+                _authHeader = Helpers.EncodeTo64($"{_username}:{_password}");
+            }
+        }
 
         #region Constructors
 
         /// <summary>
-        /// Constructor for use when providing user credentials. Users may specify if basic authorization is to be used or if more secure JWT token reliant
-        /// authorization will be used.
-        /// @Throws If LoginMode is set to TokenAad
+        /// Construct an AcrClientCredentials object from user credentials. Users may specify basic authentication or the more secure oauth2 (token) based authentication.
+        /// <exception cref="Exception"> Throws an exception if LoginMode is set to TokenAad </exception>
         /// <paramref name="mode"/> The credential acquisition mode, one of Basic, TokenAuth, or TokenAad
         /// <paramref name="loginUrl"/> The url of the registry to be used
         /// <paramref name="username"/> The username for the registry
         /// <paramref name="password"/> The password for the registry
         /// </summary>
         public AcrClientCredentials(LoginMode mode, string loginUrl, string username, string password, CancellationToken cancellationToken = default)
         {
-            _acrScopes = new Dictionary<string, string>();
-            _acrAccessTokens = new Dictionary<string, AcrAccessToken>();
-            _mode = mode;
-            if (_mode == LoginMode.TokenAad)
+            if (mode == LoginMode.TokenAad)
             {
-                throw new Exception("AAD token authorization requires you to provide the AAD_access_token");
-            }
-            // Proofing in case passed in loginurl includes https start.
-            if (loginUrl.StartsWith("https://"))
-            {
-                loginUrl.Substring("https://".Length);
-            }
-            if (loginUrl.EndsWith("/"))
-            {
-                loginUrl.Substring(0, loginUrl.Length - 1);
+                throw new ArgumentException("This constructor does not permit AAD Authentication. Please use an appropriate constructor.");
             }
 
-            _loginUrl = loginUrl;
+            _acrScopes = new Dictionary<string, string>();
```

Comment:
For the `Dictionary` I recommend being explicit about the comparer you need (ie. should it be case sensitive or not).